Client/src/test/java/org/gluu/oxauth/ws/rs/deviceauthz/DeviceAuthzFlowHttpTest.java

```diff
@@ -0,0 +1,605 @@
+
+        return tokenResponse2;
+    }
+
+    private void verifyIdToken(String idToken) throws InvalidJwtException, UnrecoverableKeyException,
+            NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+        Jwt jwt = Jwt.parse(idToken);
+        assertNotNull(jwt.getHeader().getClaimAsString(JwtHeaderName.TYPE));
+        assertNotNull(jwt.getHeader().getClaimAsString(JwtHeaderName.ALGORITHM));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.ISSUER));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.AUDIENCE));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.EXPIRATION_TIME));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.ISSUED_AT));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.SUBJECT_IDENTIFIER));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.OX_OPENID_CONNECT_VERSION));
+
+        RSAPublicKey publicKey = JwkClient.getRSAPublicKey(
+                jwksUri,
+                jwt.getHeader().getClaimAsString(JwtHeaderName.KEY_ID), clientExecutor(true));
+        RSASigner rsaSigner = new RSASigner(SignatureAlgorithm.RS256, publicKey);
+
+        assertTrue(rsaSigner.validate(jwt));
+    }
+
+    private TokenResponse processTokens(String clientId, String clientSecret, String deviceCode) {
+        TokenRequest tokenRequest = new TokenRequest(GrantType.DEVICE_CODE);
+        tokenRequest.setAuthUsername(clientId);
+        tokenRequest.setAuthPassword(clientSecret);;
+        tokenRequest.setAuthenticationMethod(AuthenticationMethod.CLIENT_SECRET_BASIC);
+        tokenRequest.setDeviceCode(deviceCode);
+
+        TokenClient tokenClient1 = newTokenClient(tokenRequest);
+        tokenClient1.setRequest(tokenRequest);
+        TokenResponse tokenResponse1 = tokenClient1.exec();
+
+        showClient(tokenClient1);
+
+        return tokenResponse1;
+    }
+
+    private void validateTokenSuccessfulResponse(TokenResponse tokenResponse) {
+        assertEquals(tokenResponse.getStatus(), 200, "Unexpected response code: " + tokenResponse.getStatus());
+        assertNotNull(tokenResponse.getEntity(), "The entity is null");
+        assertNotNull(tokenResponse.getAccessToken(), "The access token is null");
+        assertNotNull(tokenResponse.getExpiresIn(), "The expires in value is null");
+        assertNotNull(tokenResponse.getTokenType(), "The token type is null");
+        assertNotNull(tokenResponse.getRefreshToken(), "The refresh token is null");
+    }
+
+    private void assertSuccessAuthzResponse(final AuthorizationResponse authorizationResponse) {
+        assertNotNull(authorizationResponse.getCode());
+        assertNotNull(authorizationResponse.getState());
+        assertNull(authorizationResponse.getErrorType());
+    }
+
+    private void processDeviceAuthzPutUserCodeAndPressContinue(String userCode, WebDriver currentDriver, boolean complete) {
```

Comment:
Not critical but I guess we can organize selenium interaction in separate object which fits better to OOP. E.g. as in `SelectPage`.